mobile/lib/modules/home/services/asset.service.dart

```diff
@@ -1,34 +1,73 @@
 import 'dart:async';
 
 import 'package:flutter/material.dart';
+import 'package:hive/hive.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/constants/hive_box.dart';
+import 'package:immich_mobile/modules/backup/services/backup.service.dart';
+import 'package:immich_mobile/shared/models/asset.dart';
 import 'package:immich_mobile/shared/providers/api.provider.dart';
 import 'package:immich_mobile/shared/services/api.service.dart';
 import 'package:openapi/api.dart';
+import 'package:photo_manager/src/types/entity.dart';
+
+import '../../backup/models/hive_backup_albums.model.dart';
 
 final assetServiceProvider = Provider(
   (ref) => AssetService(
     ref.watch(apiServiceProvider),
+    ref.watch(backupServiceProvider),
   ),
 );
 
 class AssetService {
   final ApiService _apiService;
+  final BackupService _backupService;
 
-  AssetService(this._apiService);
+  AssetService(this._apiService, this._backupService);
 
-  Future<List<AssetResponseDto>?> getAllAsset() async {
+  Future<List<Asset>> getAllAsset() async {
+    final List<Asset> assets = [];
     try {
-      return await _apiService.assetApi.getAllAssets();
+      final Future<List<AssetResponseDto>?> remoteTask =
+          _apiService.assetApi.getAllAssets();
+
+      /* FIXME this is bad:
```

Comment:
how do we want to tackle this?

1. ignore the potential problem (currently implemented): doing this discards changes made by the background service to `HiveBackupAlbums` because the foreground app already has the data loaded
2. somehow wait until BackgroundBackup has finished (non-trivial to do): If the local cache is empty or invalid, we don't show any assets until the BackgroundBackup is done.

Reply:
If we wait for the BackgroundBackup to finish, it means the foreground UI will be locked, correct?

Reply:
I tested this with a `await Future.delayed(const Duration(seconds: 30));` and you can actually still use the app.
One issue is... depending on what the user does `getAllAsset` is called multiple times and all these calls are blocked and continue after waiting

Reply:
How does it cause problems for this PR? I think I miss somethings here to understand the problem entirely.

Reply:
found a way so that multiple places in the code can wait on the background service.

for `getAllAssets`: if the cached state is invalid, it will at first not show local assets when background service is actually running - only remote assets